src/Writers/LaTeXWriter.jl

```diff
@@ -83,19 +84,43 @@ function render(doc::Documents.Document)
                 writefooter(context, doc)
             end
             cp(STYLE, "documenter.sty")
-            if hastex()
-                outdir = joinpath(doc.user.root, doc.user.build)
-                pdf = replace("$(doc.user.sitename).pdf", " " => "")
-                try
-                    run(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $file`)
-                catch err
-                    @warn "failed to compile. Check generated LaTeX file."
-                    cp(file, joinpath(outdir, file); force = true)
+
+            str = read(file, String)
+            str = replace(str, "{Writers}\n\\end{itemize}\n\\end{itemize}" =>
+                               "{Writers}\n\\end{itemize}")
+            write(file, str)
+
+            texcmd = `latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $file`
+
+            docker = Sys.which("docker") !== nothing
+            latexmk = Sys.which("latexmk") !== nothing
+            try
+                if docker
```

Comment:
I am not sure if we should default to Docker when both are available. Also, I think it would be better if the "compiler backend" would be configured explicitly, rather then deciding which one to use automagically.